crates/ethereum/rpc/src/api_builder.rs

```diff
@@ -0,0 +1,48 @@
+use reth_evm::ConfigureEvm;
+use reth_network_api::NetworkInfo;
+use reth_provider::{CanonStateSubscriptions, FullRpcProvider};
+use reth_rpc::EthApi;
+use reth_rpc_builder::{
+    EthApiBuilder, EthApiBuilderCtx, FeeHistoryCacheBuilder, GasPriceOracleBuilder,
+};
+use reth_tasks::{pool::BlockingTaskPool, TaskSpawner};
+use reth_transaction_pool::TransactionPool;
+
+/// Ethereum layer one `eth` RPC server builder.
+#[derive(Default, Debug, Clone, Copy)]
+pub struct EthApiBuild;
+
+impl<Provider, Pool, EvmConfig, Network, Tasks, Events>
+    EthApiBuilder<Provider, Pool, EvmConfig, Network, Tasks, Events> for EthApiBuild
+where
+    Provider: FullRpcProvider,
+    Pool: TransactionPool,
+    Network: NetworkInfo + Clone,
+    Tasks: TaskSpawner + Clone + 'static,
+    Events: CanonStateSubscriptions,
+    EvmConfig: ConfigureEvm,
+{
+    type Server = EthApi<Provider, Pool, Network, EvmConfig>;
+
+    fn build(
+        &self,
+        ctx: &EthApiBuilderCtx<Provider, Pool, EvmConfig, Network, Tasks, Events>,
```

Comment:
I think we should shift this to FullNode trait and not use individual generics, given that we want to bake this into node builder anyway

Reply:
the intention is to implement the builder pattern in a next pr, which will use the `FullNodeComponents`, the problem here otherwise is cyclical dependencies between `reth-rpc-builder` and `reth-node-builder`.

I can move the implementation of trait `reth_rpc_builder::EthApiBuilder` into `reth-rpc-builder`, however, it is Layer one specific. so I'm wondering if eventually it would make more sense in the new `reth-ethereum-rpc` crate.